cmd/rpcfuzz/rpcfuzz.go

```diff
@@ -0,0 +1,1395 @@
+		Name:      "RPCTestEthGetCodeEarliest",
+		Method:    "eth_getCode",
+		Args:      []interface{}{*testContractAddress, "earliest"},
+		Validator: ValidateRegexString(`^0x$`),
+	})
+
+	// cast rpc --rpc-url localhost:8545 eth_sign "0xb9b1cf51a65b50f74ed8bcb258413c02cba2ec57" "0xdeadbeaf"
+	allTests = append(allTests, &RPCTestDynamicArgs{
+		Name:           "RPCTestEthSign",
+		Method:         "eth_sign",
+		Args:           ArgsCoinbase(ctx, rpcClient, "0xdeadbeef"),
+		Validator:      ValidateRegexString(`^0x[[:xdigit:]]{130}$`),
+		RequiresUnlock: true,
+	})
+	allTests = append(allTests, &RPCTestGeneric{
+		Name:           "RPCTestEthSignFail",
+		Method:         "eth_sign",
+		Args:           []interface{}{testEthAddress.String(), "0xdeadbeef"},
+		Validator:      ValidateError(`unknown account`),
+		IsError:        true,
+		RequiresUnlock: true,
+	})
+
+	// cast rpc --rpc-url localhost:8545 eth_signTransaction '{"from": "0xb9b1cf51a65b50f74ed8bcb258413c02cba2ec57", "to": "0x85dA99c8a7C2C95964c8EfD687E95E632Fc533D6", "data": "0x", "gas": "0x5208", "gasPrice": "0x1", "nonce": "0x1"}'
+	allTests = append(allTests, &RPCTestDynamicArgs{
+		Name:           "RPCTestEthSignTransaction",
+		Method:         "eth_signTransaction",
+		Args:           ArgsCoinbaseTransaction(ctx, rpcClient, &RPCTestTransactionArgs{To: testEthAddress.String(), Value: "0x123", Gas: "0x5208", Data: "0x", MaxFeePerGas: defaultMaxFeePerGas, MaxPriorityFeePerGas: defaultMaxPriorityFeePerGas, Nonce: "0x1"}),
+		Validator:      ValidateJSONSchema(rpctypes.RPCSchemaSignTxResponse),
+		RequiresUnlock: true,
+	})
+
+	// cast rpc --rpc-url localhost:8545 eth_sendTransaction '{"from": "0xb9b1cf51a65b50f74ed8bcb258413c02cba2ec57", "to": "0x85dA99c8a7C2C95964c8EfD687E95E632Fc533D6", "data": "0x", "gas": "0x5208", "gasPrice": "0x1", "nonce": "0x1"}'
+	allTests = append(allTests, &RPCTestDynamicArgs{
+		Name:           "RPCTestEthSendTransaction",
+		Method:         "eth_sendTransaction",
+		Args:           ArgsCoinbaseTransaction(ctx, rpcClient, &RPCTestTransactionArgs{To: testEthAddress.String(), Value: "0x123", Gas: "0x5208", Data: "0x", MaxFeePerGas: defaultMaxFeePerGas, MaxPriorityFeePerGas: defaultMaxPriorityFeePerGas}),
+		Validator:      ValidateRegexString(`^0x[[:xdigit:]]{64}$`),
+		RequiresUnlock: true,
+	})
+
+	// cast rpc --rpc-url localhost:8545 eth_sendRawTransaction '{"from": "0xb9b1cf51a65b50f74ed8bcb258413c02cba2ec57", "to": "0x85dA99c8a7C2C95964c8EfD687E95E632Fc533D6", "data": "0x", "gas": "0x5208", "gasPrice": "0x1", "nonce": "0x1"}'
+	allTests = append(allTests, &RPCTestDynamicArgs{
+		Name:      "RPCTestEthSendRawTransaction",
+		Method:    "eth_sendRawTransaction",
+		Args:      ArgsSignTransaction(ctx, rpcClient, &RPCTestTransactionArgs{To: testEthAddress.String(), Value: "0x123", Gas: "0x5208", Data: "0x", MaxFeePerGas: defaultMaxFeePerGas, MaxPriorityFeePerGas: defaultMaxPriorityFeePerGas}),
+		Validator: ValidateRegexString(`^0x[[:xdigit:]]{64}$`),
+	})
```

Comment:
niceee! in our previous implementation, we validated for correct balance changes of gas * gasPrice + amount sent. would it make sense to add it here or to put that at an EVM lower level test?

Reply:
It's a good question. I think it can't hurt. We could have some state / account level validators and chain them together with a `RequireAll`.